src/styles/app-tooltip/AppTooltip.tsx

```diff
@@ -0,0 +1,87 @@
+import { KeyboardEvent, forwardRef, useRef, useState } from "react";
+
+import classNames from "classnames";
+import "design-system/app-tooltip/AppTooltip.scss";
+
+import { AppTooltipProps } from "@/styles/app-tooltip/AppTooltip.types.ts";
+import AppTypography from "@/styles/app-typography/AppTypography.tsx";
+
+const AppTooltip = forwardRef<HTMLDivElement, AppTooltipProps>(function (
+  {
+    position = "top",
+    width = "nowrap",
+    content,
+    children,
+    className,
+    ...props
+  },
+  ref
+) {
+  const [isVisible, setIsVisible] = useState(false);
+  const timeoutRef = useRef<number | null>(null);
```

Comment:
```suggestion
  const timeoutRef = useRef<Timer | null>(null);
```

But there is an error in global types (src/types), please replace as follows:

```tsx
export type Timer = ReturnType<typeof setTimeout>;
```